Models/getModels.js

```diff
@@ -16,13 +17,61 @@ exports.selectCategories = () => {
     });
 };
 
-exports.selectReviews = () => {
-  return db
-    .query(
-      `
-      SELECT *, COUNT(review_id) AS comment_count FROM reviews group by review_id;
-  `,
-    )
+exports.selectReviews = (categories, sortBy, order, query) => {
+  return doesCategoryExist(categories)
+    .then(() => {
+      const sortAllowed = [
+        "owner",
+        "title",
+        "review_id",
+        "category",
+        "review_img_url",
+        "review_body",
+        "created_at",
+        "votes",
+        "designer",
+        "comment_count",
+        "default",
+      ];
+      const orderAllowed = ["asc", "desc"];
+
+      const queries = [];
+      let queryString =
+        "SELECT *, COUNT(review_id) AS comment_count FROM reviews ";
+
+      if (categories) {
+        queryString += "WHERE category = $1 ";
+        queries.push(categories);
+      }
+
+      queryString += "GROUP BY review_id ";
+
+      if (sortBy) {
+        if (!sortAllowed.includes(sortBy)) {
+          return Promise.reject({ status: 400, message: "invalid query" });
+        } else {
+          if (sortBy !== "default") {
```

Comment:
You shouldn't allow default as an option. The default option of 'created_at' is when they don't provide a query at all.